inspire/testsuite/test_bibtex.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import pkg_resources
+import os
+
+from dojson.contrib.marc21.utils import create_record
+
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+from inspire.dojson.hep import hep
+from invenio.base.wrappers import lazy_import
+Bibtex = lazy_import('inspire.utils.bibtex.Bibtex')
+
+
+class BibtexTests(InvenioTestCase):
+
+    def setUp(self):
+        self.marcxml = pkg_resources.resource_string('inspire.testsuite',
+                                                     os.path.join(
+                                                         'fixtures',
+                                                         'test_hep_formats.xml')
+                                                     )
+        record = create_record(self.marcxml)
+
+        self.hep_record = hep.do(record)
+
+        self.sample_bibtex = {
+            'collections': ('article', 'article'),
```

Comment:
Because _get_entry_type returns the 2 things, the entry_type and the
original_entry. So in my record the original and the entry are the same.

2015-10-29 10:37 GMT+01:00 Javier Martin Montull <notifications@github.com>:

> In inspire/testsuite/test_bibtex.py
> <https://github.com/inspirehep/inspire-next/pull/445#discussion_r43364774>
> :
>
> > +
> > +
> > +class BibtexTests(InvenioTestCase):
> > +
> > +    def setUp(self):
> > +        self.marcxml = pkg_resources.resource_string('inspire.testsuite',
> > +                                                     os.path.join(
> > +                                                         'fixtures',
> > +                                                         'test_hep_formats.xml')
> > +                                                     )
> > +        record = create_record(self.marcxml)
> > +
> > +        self.hep_record = hep.do(record)
> > +
> > +        self.sample_bibtex = {
> > +            'collections': ('article', 'article'),
>
> why are the collections repeated?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/445/files#r43364774>.
>